corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -521,27 +543,27 @@ hqDefine("data_dictionary/js/data_dictionary", [
             return pattern.test(nameStr);
         }
 
-        self.newPropertyNameValid = ko.computed(function () {
-            if (!self.newPropertyName()) {
+        self.newPropertyNameValid = function (name) {
```

Comment:
What was the motivation for changing this from a computed property?

Reply:
`dataDictionaryModel.newPropertyName()` is no longer the value that I need to test. I had a choice:

1. Change `newPropertyNameValid()` and `newPropertyNameUnique()` to functions that I can pass `name` to.

2. Move them to `groupsViewModel` where `self.newPropertyName()` will be the same. But then I would either need to duplicate `isNameValid()`, or create a new property on `groupsViewModel` that refers to `dataDictionaryModel` so that I can call `dataDictionaryModel.isNameValid()`.

The first option seemed simpler.

Reply:
Thanks for the clarification! I agree option (1) sounds right.